cmd/runtimecfg/node-ip.go

```diff
@@ -138,33 +138,52 @@ func set(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func checkAddressUsable(chosen []net.IP) (err error) {
+	// If using IPv6, verify that the choosen address isn't tentative
+	// i.e. we can actually bind to it
+	if len(chosen) > 0 && net.IPv6len == len(chosen[0]) {
+		_, err = net.Listen("tcp", "["+chosen[0].String()+"]:")
+		if err != nil {
+			log.Errorf("Chosen node IP is not usable")
+			return err
+		}
+	}
+	return err
+}
+
 func getSuitableIPs(retry bool, vips []net.IP, preferIPv6 bool) (chosen []net.IP, err error) {
 	// Enable debug logging in utils package
 	utils.SetDebugLogLevel()
 	for {
 		if len(vips) > 0 {
 			chosen, err = utils.AddressesRouting(vips, utils.ValidNodeAddress)
 			if len(chosen) > 0 || err != nil {
-
-				// If using IPv6, verify that the choosen address isn't tentative
-				// i.e. we can actually bind to it
-				if len(chosen) > 0 && net.IPv6len == len(chosen[0]) {
-					_, err := net.Listen("tcp", "["+chosen[0].String()+"]:")
-					if err != nil {
-						log.Errorf("Chosen node IP is not usable")
-						if !retry {
-							return nil, fmt.Errorf("Failed to find node IP")
-						}
-						time.Sleep(time.Second)
-						continue
+				if err == nil {
+					err = checkAddressUsable(chosen)
+				}
+				if err != nil {
+					if !retry {
+						return nil, fmt.Errorf("Failed to find node IP")
 					}
+					time.Sleep(time.Second)
+					continue
 				}
 				return chosen, err
 			}
 		}
 		if len(chosen) == 0 {
 			chosen, err = utils.AddressesDefault(preferIPv6, utils.ValidNodeAddress)
 			if len(chosen) > 0 || err != nil {
+				if err == nil {
```

Comment:
This logic looks similar to line 164 and so on. Should we convert to a helper function to avoid dup code?

Reply:
Same problem. Things get weird if we move more of this logic into the function because we end up needing some way to communicate flow control back into the loop here. It is a bit duplicated, but I think the code is more clear than if we try to deduplicate this any further.